### PR TITLE
Fix max instance

### PR DESCRIPTION
**Pull request by cheukt**
put maxInstance on subtype and then also doublecheck if resource is a remote when tallying up instances

### CODE REVIEW

**Review by bhaney — COMMENTED** [review]
thanks! LGTM

**Review by bhaney — APPROVED** [review]

File: robot/impl/local_robot.go [hunk 1]
```diff
@@ -776,14 +778,14 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
...
+// checkMaxInstance checks to see if the local robot has reached the maximum number of a specific service type that are local.
+func (r *localRobot) checkMaxInstance(subtype resource.Subtype, max int) error {
 	maxInstance := 0
 	for _, n := range r.ResourceNames() {
-		if n.Subtype == name.Subtype {
+		if n.Subtype == subtype && !n.ContainsRemoteNames() {
 			maxInstance++
-			if maxInstance == f.MaxInstance {
-				return errors.Errorf("Max instance number reached for service type: %s", name.Subtype)
+			if maxInstance == max {
```
Comment (stuqdog) [inline robot/impl/local_robot.go:787]: I would have thought we'd want to return err only if `maxInstance > max`.

File: robot/impl/local_robot_test.go [hunk 2]
```diff
@@ -1766,3 +1766,59 @@ func TestCheckMaxInstanceInvalid(t *testing.T) {
...
+
+	maxInstance := 0
+	for _, name := range r.ResourceNames() {
+		if name.Subtype == datamanager.Subtype {
+			maxInstance++
+		}
+	}
+	test.That(t, maxInstance, test.ShouldEqual, 2)
+
+	_, err = r.ResourceByName(datamanager.Named("remote:builtin"))
```
Comment (stuqdog) [inline robot/impl/local_robot_test.go:1822]: Should we add a test here for the case where we go above max and expect an error?

**Review by stuqdog — CHANGES_REQUESTED** [review]

File: robot/impl/local_robot.go (hunk 1 above)
Comment (cheukt) [reply robot/impl/local_robot.go:787]: in this case, we're checking the number of existing instances when creating a new instance - so equal means we should err

File: robot/impl/local_robot_test.go (hunk 2 above)
Comment (cheukt) [reply robot/impl/local_robot_test.go:1822]: will do

Comment (stuqdog) [reply robot/impl/local_robot_test.go:1822]: Ah gotcha, okay. I think that is a little unclear, ideally I think it'd be helpful to have a test case where we expect an error and see it, but I think it's not a big enough deal to delay merging if you want to get this out quickly.

**Review by stuqdog — APPROVED** [review]

**Comment by cheukt** [comment]
closing since I can't update the branch https://github.com/viamrobotics/rdk/pull/1348

**Pull request closed without merging**